lib/checks/text-alternatives/link-text.js

```diff
@@ -0,0 +1,29 @@
+function getText(elem) {
+	let returnObj = {};
+
+	returnObj['link-text'] = elem.innerText;
+	returnObj['link-title'] = elem.title;
+	returnObj['link-href'] = elem.href;
+
+	return returnObj;
+}
+
+let nodeText = getText(node);
+
+var elems = document.body.querySelectorAll('a[href]');
+
+// Loop trough all the elements to check against
+for (var i = 0; i < elems.length; i++) {
```

Comment:
Using a `for` loop here, leads to redundant iterations.
Use `.some`, to exit iteration on first match.

Reference:
[MDN Array Some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)